doc/src/bond_cpp.md

```diff
@@ -185,6 +185,65 @@ particular protocol reader:
 
 See example: `examples/cpp/core/marshaling`.
 
+Schema evolution
+================
+
+Bond does not use explicit versioning to deal with changes to schemas (and the
+resulting types) over time. Instead, Bond supports certain schema evolution
+operations which allow the producer and consumer of Bond types to evolve
+without lockstep coordination.
+
+The following changes to a schema will never break compatibility across the wire:
+
+- Adding or removing an `optional` or `required_optional` field
+- Changing a field's type between `int32` and `enum`
+- Changing a field's type between `vector<T>` and `list<T>`
+- Changing a field's type between `blob` and `vector<uint8>` or `blob` and
+  `list<uint8>`
+- Changing a field's type between `T` and `bonded<T>`
+- Adding new enumeration constants that don't alter existing constants (beware
+  of implicit reordering)
+
+The following changes to a type are generally safe but require some
+consideration about how the change is rolled out:
+
+- Changing a field between `optional` and `required_optional` or between
+  `required_optional` and `required`. The `required_optional`. modifier
+  facilitates a two-step process for changing between `optional` and
+  `required`, but the first step must be completed on both the producer and the
+  consumer sides before the second step can be started on either side.
+- Promoting a field with a numeric type from a smaller size to a larger size
+  (e.g.: `float` to `double`, `uint8` to `uint16`, `uint8` to `uint32`, `int8`
+  to `int16`, etc.). The consumer must get the change before the producer.
+  Changing from unsigned to signed or vice versa is NOT compatible (e.g.:
+  `uint8` to `int16`).
+- Promoting from `int8` or `int16` to `enum`. The consumer must get the change
+  before the producer.
+
+These following changes will break wire compatibility and are not recommended:
+
+- Adding or removing `required` fields
+- Incompatible change of field types (any type change NOT covered above); e.g.:
+  `int32` to `string`, `string` to `wstring`
+- Changing of field ordinals
+- Changing of inheritance hierarchy (add/remove/substituting base struct)
+- Changing between `required` and `optional` directly
+- Changing the default value of a field
+- Changing existing enumeration constants in any way (including implicit
+  renumbering)
+
+Some best practices and other considerations to keep in mind:
+
+- When removing a field, comment it out rather than removing it altogether so
+that the field ordinal is not reused in future edits of the schema
+- When working with untagged protocols like SimpleBinaryProtocol, great care
```

Comment:
Might want to mention that runtime SchemaDefs can be used with a link to the SimpleBinaryProtocol section.